tds.go

```diff
@@ -995,43 +1175,64 @@ initiate_connection:
 	}
 
 	// processing login response
-	reader := startReading(&sess, ctx, nil)
-	for {
-		tok, err := reader.nextToken()
-		if err == nil {
-			if tok == nil {
-				break
-			} else {
-				switch token := tok.(type) {
-				case sspiMsg:
-					sspi_msg, err := auth.NextBytes(token)
-					if err != nil {
-						return nil, err
-					}
-					if len(sspi_msg) > 0 {
-						outbuf.BeginPacket(packSSPIMessage, false)
-						_, err = outbuf.Write(sspi_msg)
+
+	for pending := true; pending; {
+		reader := startReading(&sess, ctx, nil)
+		for {
```

Comment:
Why did you add another loop above this one? This concerns me a bit.

Reply:
The login sequence for federated authentication (and I think SSPI) spans multiple packets: we need to loop here as the startReading() call was recently refactored to assume a single response. Before that this loop was implicit because the call to processResponses was used that kept reading packets and we could just send the additional login info when needed.

Reply:
Okay that makes sense.